### ISSUE TITLE
npm run build results in different hashes depending on current working directory

### ISSUE THREAD

**Issue by stefanaxelsson** [vercel/next.js#24116]
### What version of Next.js are you using?

Tested: 10.0.6, 10.1.3

### What version of Node.js are you using?

12.21

### What browser are you using?

Chrome

### What operating system are you using?

Mac OS / docker image: node:12.21.0

### How are you deploying your application?

Docker

### Describe the Bug

Running `ASSET_PREFIX=true npm run build` in different working directories results in different hashes for js files, even though the content is the same. The bug seems to have been introduced in 10.0.6 since 10.0.5 works.

I also noted that the the working directory is saved as appDir in required-server-files.json.

### Expected Behavior

app.js to have the same hash when content and build id is the same.

### To Reproduce

Clone https://github.com/stefanaxelsson/test-app
Run `nvm use`
Run `npm ci`
Run `ASSET_PREFIX=true npm run build`

Copy the working directory and give it a new name
Run `nvm use; ASSET_PREFIX=true npm run build` in the new directory. Note that hash for app.js has changed even though the content is the same.

Alternatively using docker:
Clone https://github.com/stefanaxelsson/test-app
Run `docker build . -f ./Dockerfile --tag server:1`
Run `docker build . -f ./Dockerfile-cdn --tag cdn:1`

Note that hash for app.js has changed even though the content is the same and the only thing that differs between the images is the working directory.

**Comment by davscro** [comment]
@stefanaxelsson , do you have any rewrites specified in next.config?

**Comment by stefanaxelsson** [comment]
> @stefanaxelsson , do you have any rewrites specified in next.config?

No, you can see the [next.config here](https://github.com/stefanaxelsson/test-app/blob/main/next.config.js). It uses a static build id to ensure it's the same between builds and a assetPrefix to simulate the usage of cdn.

**Comment by KarloDuderija** [comment]
The bug appears right after version 10.0.5 and is already present in 10.0.6.-canary.0. 
I also have the same problem.

**Comment by jakobrahm** [comment]
Any solution in sight? We're stuck on 10.0.5.

**Comment by tunesmith** [comment]
This bug has nailed us in production, and it doesn't depend on different working directories, either. Here's an alternative reproduction:

1. Clone the above git repo in the bug description
2. Change next.config.js to this:

**Comment by fabb** [comment]
The webpack 5 migration guide suggests:
> When using [hash] placeholder in webpack configuration, consider changing it to [contenthash]. It is not the same, but proven to be more effective.
> https://webpack.js.org/migrate/5/

Could it be that next.js internally uses `[hash]` instead of `[contenthash]`?

**Comment by tunesmith** [comment]
Is this fixed by #25055 ?

**Comment by davscro** [comment]
yes, it is fixed by #25055

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
